File: clients/chat-room/docker/nodejs/src/producer.ts
import { Kafka } from "kafkajs";
const redpanda = new Kafka({
  brokers: ["localhost:19092"],
});
const producer = redpanda.producer();
export async function getConnection(user: string) {
  try {
    await producer.connect();
    return async (message: string) => {
      await producer.send({
        topic: "chat-room",
        messages: [{ value: JSON.stringify({ message, user }) }],
      });
    };
  } catch (error) {
    console.error("Error:", error);
  }
}
export async function disconnect() {
  try {
    await producer.disconnect();
  } catch (error) {
    console.error("Error:", error);
  }
}

File: clients/chat-room/docker/go/producer.go
package main
import (
    "context"
    "encoding/json"
    "github.com/twmb/franz-go/pkg/kgo"
)
type Producer struct {
    client *kgo.Client
    topic  string
}
func NewProducer(brokers []string, topic string) *Producer {
    client, err := kgo.NewClient(
        kgo.SeedBrokers(brokers...),
    )
    if err != nil {
        panic(err)
    }
    return &Producer{client: client, topic: topic}
}
func (p *Producer) SendMessage(user, message string) {
    ctx := context.Background()
    msg := Message{User: user, Message: message}
    b, _ := json.Marshal(msg)
    p.client.Produce(ctx, &kgo.Record{Topic: p.topic, Value: b}, nil)
}
func (p *Producer) Close() {
    p.client.Close()
}